doc/expectations.md

```diff
@@ -0,0 +1,279 @@
+# Expectations
+
+This framework allows problem author to express their expecations for the behaviour of a submission on the test data.
+
+## Test Case Verdict
+
+The behaviour of a submission on a _single_ test case is summarised in a *verdict*.
+
+The verdicts are:
+
+* AC: Accepted. The submission terminates successfully within the time limit and the output vaidator accepts the submission output.
+* WA: The submission terminates successfully within the time limit, but the output validator rejects the submission output on this testcase.
+* TLE: The submission does not terminate within the time limit.
+* RTE: The submission aborts within the time limit with a runtime error.
+
+
+## Common Expectations for a Submission
+
+
+The expected behaviour of a submission on the test data often falls into a number of common classes:
+
+* accepted: Every test case is `AC`.
+* wrong answer: Every test case receives `AC` or `WA`;  _some_ test case receives `WA`.
+* time limit exceeded: Every test case receives `AC` or `TLE`;  _some_ test case receives `TLE`.
+* runtime exception: Every test case receives `AC` or `RTE`;  _some_ test case receives `RTE`.
```

Comment:
I **will** trip over `time <space> limit` vs `run <no space> time` all the time. oh well. Let's make nice IDE integration to autocomplete all options here :)

Reply:
(These details are not really important to me, by the way and shouldn’t get in the way of anything.)